### PR TITLE
patch tcl for CVE-2023-36328

### PR DESCRIPTION
**Pull request by bfjelds**
###### Merge Checklist  

**All** boxes should be checked before merging the PR *(just tick any boxes which don't apply to this PR)*
- [ ] The toolchain has been rebuilt successfully (or no changes were made to it)
- [ ] The toolchain/worker package manifests are up-to-date
- [ ] Any updated packages successfully build (or no packages were changed)
- [ ] Packages depending on static components modified in this PR (Golang, `*-static` subpackages, etc.) have had their `Release` tag incremented.
- [ ] Package tests (%check section) have been verified with RUN_CHECK=y for existing SPEC files, or added to new SPEC files
- [ ] All package sources are available
- [ ] cgmanifest files are up-to-date and sorted (`./cgmanifest.json`, `./toolkit/scripts/toolchain/cgmanifest.json`, `.github/workflows/cgmanifest.json`)
- [ ] LICENSE-MAP files are up-to-date (`./SPECS/LICENSES-AND-NOTICES/data/licenses.json`, `./SPECS/LICENSES-AND-NOTICES/LICENSES-MAP.md`, `./SPECS/LICENSES-AND-NOTICES/LICENSE-EXCEPTIONS.PHOTON`)
- [ ] All source files have up-to-date hashes in the `*.signatures.json` files
- [ ] `sudo make go-tidy-all` and `sudo make go-test-coverage` pass
- [ ] Documentation has been updated to match any changes to the build system
- [ ] Ready to merge

---

###### Summary 
Adapt nist linked patch to libtommath embedded in tcl to address CVE-2023-36328

###### Change Log  
- CVE-2023-36328

###### Does this affect the toolchain?  
**NO**

###### Links to CVEs  
- https://nvd.nist.gov/vuln/detail/CVE-2023-36328

###### Test Methodology

- https://dev.azure.com/mariner-org/mariner/_build/results?buildId=418862&view=logs&j=7909073b-e471-53c1-2c44-627e2efc0564&t=7909073b-e471-53c1-2c44-627e2efc0564

### CODE REVIEW

File: SPECS/tcl/tcl.spec [hunk 1]
```diff
@@ -31,6 +32,7 @@ Headers and development libraries for tcl
 
 %prep
 %setup -q -n %{name}%{version}
+%patch0 -p1
 
 %build
 cd unix
```
Comment (bfjelds) [inline SPECS/tcl/tcl.spec:38]: add changelog entry

Comment (bfjelds) [reply SPECS/tcl/tcl.spec:38]: new version too

File: SPECS/tcl/tcl.spec [hunk 2]
```diff
@@ -31,6 +32,7 @@ Headers and development libraries for tcl
 
 %prep
 %setup -q -n %{name}%{version}
+%patch0 -p1
```
Comment (mfrw) [inline SPECS/tcl/tcl.spec:35]: How about we use `%autosetup` :)
```suggestion
%autosetup -n %{name}%{version} -p1
```

**Review by mfrw — APPROVED** [review]
LGTM :)

While we are here .. we might as well fix line `10`:
`s/http/https/`

We would always prefer `https` urls instead of `http`
This would mean an update to the `cgmanifest` entry as well :)

File: SPECS/tcl/tcl.spec (hunk 2 above)
Comment (bfjelds) [reply SPECS/tcl/tcl.spec]: updated for suggestions in commit below.

**Review by mfrw — APPROVED** [review]

**Pull request merged**